### ISSUE TITLE
Unknown ASCII Armor type

### ISSUE THREAD

**Issue by stag-enterprises** [mailvelope/mailvelope#865]
When receiving a encrypted Email and viewing it through Roundcube, I get the error "Could not decrypt this message. Unknown ASCII Armor Error". However, when copying the contents to Mailvelope's decryptor in the extension dashboard, it decrypts perfectly fine. If it helps, I have API integration turned on for Roundcube.

**Comment by stag-enterprises** [comment]
Just tried using Roundcube without API integration, it works fine.

**Comment by rimmer** [comment]
I tried with Roundcube 1.7 and I couldn't reproduce the issue (with Mailvelope 6.0). 
What version of Roundcube do you use?
What version of Mailvelope?

Is it a specific email or it happens for all encrypted emails? If the former, maybe you can share a sample of the email, that would be precious! Of course, I mean a sample that doesn't contain any personal or information (perhaps just sending a mail to yourself).

**Comment by stag-enterprises** [comment]
@rimmer I'm currently on Roundcube 1.6.9, and Mailvelope 6.0.0

It seems to be happening for all emails, as far as I can tell. Here's an example of an email I received, it gives the “Unknown ASCII Armor type” error (zipped because GitHub doesn't allow EML files). 
[sourcehut introduction.eml.zip](https://github.com/user-attachments/files/18046147/sourcehut.introduction.zip)

**Comment by rimmer** [comment]
> @rimmer I'm currently on Roundcube 1.6.9, and Mailvelope 6.0.0
> 
> It seems to be happening for all emails, as far as I can tell. Here's an example of an email I received, it gives the “Unknown ASCII Armor type” error (zipped because GitHub doesn't allow EML files). [sourcehut introduction.eml.zip](https://github.com/user-attachments/files/18046147/sourcehut.introduction.zip)

Thanks for the sample. I gave a shot in Roundcube 1.7 and it appears to be working properly giving me an error that I don't have a private key to decrypt, meaning that at least reading of the encrypted email works fine. 

![image](https://github.com/user-attachments/assets/bf597437-a9de-4531-a4d6-c875abddbf4b)

Are you able to test on Roundcube 1.7? You can try spinning off `roundcube/roundcubemail` locally with the following docker environment (example):
```bash
docker run -e ROUNDCUBEMAIL_DEFAULT_HOST=ssl://<your IMAP host> -e ROUNDCUBEMAIL_SMTP_SERVER=ssl://<your SMTP host> -e ROUNDCUBEMAIL_DEFAULT_PORT=993 -e ROUNDCUBEMAIL_SMTP_PORT=587 -p 8000:80 -d roundcube/roundcubemail
```
of course if know your server credentials.

**Comment by stag-enterprises** [comment]
Unfortunately, I'm not able to try Roundcube 1.7 since I don't own the mail server. I can try to ask the admins to update it though, I'll see if it can be done.

**Comment by stag-enterprises** [comment]
@rimmer I might be missing something really obvious here, but I don't see any [mention of version 1.7](https://github.com/roundcube/roundcubemail/releases)? The latest version I see is 1.6.9

**Comment by rimmer** [comment]
> @rimmer I might be missing something really obvious here, but I don't see any [mention of version 1.7](https://github.com/roundcube/roundcubemail/releases)? The latest version I see is 1.6.9

Right, I am sorry, I was testing against unreleased nighty version. I just got the version from my Roundcube about page and didn't notice it hasn't been released. Let me check against 1.6.9.

**Comment by rimmer** [comment]
So, I checked with 1.6.9 and it seem to be working. The sample you provided is a PGP/MIME message (see more in [this article](https://proton.me/support/pgp-mime-pgp-inline)). I tried sending PGP/MIME to myself and it was working too.

1. Do you mostly receive PGP/MIME messages? If you're not sure, then do you receive them from one source mostly (I'm asking because you said you have this issue with all of them)?
2. Do you know what software was used to create the message you've got? The reason I'm asking is because the sample you provided does not follow the common structure of PGP/MIME, namely `Content-Type` doesn't contain `name` and `Content-Disposition` header doesn't exist.

**Comment by stag-enterprises** [comment]
1) Yes and yes
2) Not sure

Weird that it works for you. I tried sending myself an email using Roundcube's builtin encrypt email feature, and it sent fine but was still unable to decrypt. (Decrypted fine in the dash though)

**Comment by meaz** [comment]
I can reproduce this also:
With API on, I can write encrypted email without issue. But I can not read them as i have the same error: "Could not decrypt this message. Unknown ASCII Armor Error".
With API off, I can read those emails without problems.
To be clear, I use Roundcube to send myself that encrypted email that I can not read on Roundcube. I can read on Evolution for example. Or with API off as I already wrote.

I use roundcube 1.6.9

**Comment by austinhuang0131** [comment]
Roundcube 1.6.9 Mailvelope 5.2.0 Firefox

The armored message passed to `onSetArmored` is like
```
This is a PGP/MIME encrypted message. You can decrypt the message with addons like Mailvelope to read it in our webmail client.-----BEGIN PGP MESSAGE-----

<crypt text>
-----END PGP MESSAGE-----
```
and since `-----BEGIN PGP MESSAGE-----` is [not on its own line](https://github.com/openpgpjs/openpgpjs/blob/965e63b672aeeb4ce483d14162c28f2715772251/src/encoding/armor.js#L37), decryption with API fails.
